services/todo-legacy/test/controller/todo-controller.test.ts

```diff
@@ -0,0 +1,121 @@
+
+      stub.find.withArgs(null, sinon.match({
+        where: sinon.match({ title: 'test2'}),
+      })).returns(fakeData[1]);
+      let result = await controller.getTodo('test2');
+      expect.equal(result, fakeData[1], 'data was correct');
+
+    });
+  });
+
+  describe('createTodo', () => {
+    it('returns the object after creating it', async () => {
+      let stub = sinon.createStubInstance(TodoRepository);
+      controller.repository = stub;
+      stub.create.withArgs(fakeData[0]).returns(fakeData[0]);
+      let result = await controller.createTodo(fakeData[0]);
+      expect.equal(result, fakeData[0], 'data was correct');
+    });
+  });
+
+  describe('replaceTodo', () => {
+    it('returns the replaced item', async () => {
+      let stub = sinon.createStubInstance(TodoRepository);
+      controller.repository = stub;
+      let replacement = {
+        title: 'foo',
+        body: 'bar',
+      };
+      let expected = _.merge({ id: 1 }, replacement);
+      stub.replace.withArgs(1, replacement)
+        .returns(expected);
+      let result = await controller.replaceTodo(1, replacement);
+      expect.equal(result, expected, 'data was correct');
+
+    });
+  });
+
+  describe('updateTodo', () => {
+    it('returns the updated version of the object', async () => {
+      let stub = sinon.createStubInstance(TodoRepository);
+      controller.repository = stub;
+      let replacement = {
+        title: 'foo',
+      };
+      let expected = _.merge({ id: 1 }, replacement);
+      stub.update.withArgs(1, replacement)
+        .returns(expected);
+      let result = await controller.updateTodo(1, replacement);
+      expect.equal(result, expected, 'data was correct');
+    });
+    // There's no unhappy path tests here for missing ID, because
+    // it's handled at the repository layer, which we are stubbing!
+  });
+
+  describe('deleteTodo', () => {
+    it('works on one item', async () => {
+      let stub = sinon.createStubInstance(TodoRepository);
+      controller.repository = stub;
+      // The null filter is automatically replaced with an empty object in
+      // controller layer!
+      let expected = { count: 1 };
+      stub.delete.withArgs(1).returns(expected);
+      let result = await controller.deleteTodoById(1);
+      expect.equal(result, expected, 'data was correct');
+    });
+
+    it('can filter by title', async () => {
+      let stub = sinon.createStubInstance(TodoRepository);
+      controller.repository = stub;
+      // Use a known-good implementation to retrieve our expected result.
+      let expected = _.filter(fakeData, { title: 'test2' });
+      stub.delete.withArgs(null, sinon.match({
+        where: sinon.match({ title: 'test2' }),
+      })).returns({ count: expected.length });
+      let result = await controller.deleteTodo('test2');
+      expect.exists(result, 'result exists');
+      expect.equal(result.count, expected.length, 'count was correct');
+    });
+  });
```

Comment:
Ditto for the above, don't test your own mock/fake data, just assert the methods were called properly with the correct arguments.